lib/guardian/category_guardian.rb

```diff
@@ -46,6 +46,11 @@ def cannot_delete_category_reason(category)
     nil
   end
 
+  def can_see_serialized_category?(category_id:, read_restricted:)
+    return true if !read_restricted
+    secure_category_ids.include?(category_id)
+  end
```

Comment:
@SamSaffron I've simplified the check here as per what the original query was doing.

https://github.com/discourse/discourse/blob/cd6ab7bdd78d1709f55d9fd0938d59c4660e1767/app/models/category.rb#L110-L118